src/python/vdo-setup-dmtest/vdo-setup-dmtest.py

```diff
@@ -0,0 +1,512 @@
+      msg = "This script must be run by a user with sudo priviledges"
+      raise PermissionError(msg)
+
+    # Get the configuration
+    self.getConfig(configFile = self.configFile)
+    self.log.info("Configuration file read and processed")
+
+    # Set workspace parameter if not defined in the configuration
+    if not self.config['WORKSPACE']:
+      self.config['WORKSPACE'] = os.path.join(RUN_DIR, 'repos')
+
+    # Find all repo keys in the config object and store them in the self.repoKeys list
+    self.getRepoKeys()
+    self.log.debug("Repo keys found: {0}".format(self.repoKeys))
+
+    # Verify the configuration device parameters
+    self.verifyDevices()
+
+    # Verify self.config['LINUX_SOURCE'] exists and is defined
+    self.verifyLinuxSource()
+
+  def parseArgs(self, args):
+    """
+    Parse any input arguments and apply defaults
+    """
+    parser = argparse.ArgumentParser(
+      prog='vdo-setup-dmtest',
+      description='Installs and sets up a dmtest-python environment for running vdo tests.',
+      epilog='For additional information see README.md within the module directory.')
+    parser.add_argument(
+      "--config",
+      action = "store",
+      dest = "configFile",
+      default = os.path.join(RUN_DIR,'config.yaml'),
+      help = "The full path to the configuration YAML file. Default is './config.yaml'.")
+    parser.add_argument(
+      "--debug",
+      action = "store_true",
+      dest = "debug",
+      default = False,
+      help = "Turns on debug logging to the log file.")
+    parser.add_argument(
+      "--returnEnv",
+      action = "store_true",
+      dest = "returnEnv",
+      default = False,
+      help = "Turns on returning a list of environment variables that need to be exported" \
+             + " upon successful completion.")
+    parser.add_argument(
+      "--cleanLog",
+      action = "store",
+      type = int,
+      dest = "numDays",
+      default = 7,
+      help = "The number of days before recycling the log file. Default is 7.")
+    parser.add_argument(
+      "--runlog",
+      action = "store",
+      dest = "logFile",
+      default = os.path.join(RUN_DIR, 'run.log'),
+      help = "The full path to the log file for logging output. Default is './run.log'.")
+    return parser.parse_known_args(args)
+
+  def getConfig(self, configFile):
+    """
+    Read the config file and store the name value pairs in a dictionary
+    """
+    self.log.debug("Reading the config file at {0}".format(configFile))
+    if not os.path.exists(configFile):
+      self.log.error("Unable to read config file at {0}: No such file".format(configFile))
+      sys.exit(1)
+
+    with open(configFile, 'r') as file:
+      loadedConfig = yaml.safe_load(file)
+
+    if not bool(loadedConfig):
+      self.log.error("Unable to load config file ({0})".format(configFile))
+      return False
+
+    for key in loadedConfig.keys():
+      self.config[key] = loadedConfig[key]
```

Comment:
Why not just use something like `self.config = loadedConfig`?  It doesn't appear to me that you're using self.config anywhere else before this.

Reply:
Fixed in latest push.